src/dungeon.cpp

```diff
@@ -1299,35 +1299,38 @@ static void commandQuit() {
     }
 }
 
-static void commandPreviousMessage() {
-    int maxMessages = MESSAGE_HISTORY_SIZE;
+static int8_t calculateMaxMessageCount () {
+    int8_t max_messages = MESSAGE_HISTORY_SIZE;
 
     if (command_count > 0) {
-        maxMessages = command_count;
-        if (maxMessages > MESSAGE_HISTORY_SIZE) {
-            maxMessages = MESSAGE_HISTORY_SIZE;
+        max_messages = command_count;
```

Comment:
Not sure it's a good idea to be assigning an `int` to an `uint8_t`. Perhaps utilise the if statement here.